File: src/Supplier.java
// Постачальник: залежність від проєкту, постачає матеріали
public class Supplier extends Participant {
    private String companyName;

    public Supplier(int id, String name, String companyName) {
        super(id, name);
        this.companyName = companyName;
        System.out.println("[Supplier] Created Supplier: " + name + ", company=" + companyName);
    }

    // Залежність: постачальник залежить від проєкту під час доставки
    public void deliverMaterials(ConstructionProject project) {
        System.out.println("[Supplier] Delivering materials for project: " + project.getTitle());
    }

    @Override
    public String getRole() {
        return "Supplier";
    }
}

File: construction/python/Supplier.py
class Supplier(Participant):
    def __init__(self, id: int, name: str, company_name: str):
        super().__init__(id, name)
        self._company_name = company_name
        print(f"[Supplier] Created Supplier: {name}, company={company_name}")

    # Залежність: постачальник залежить від проєкту під час доставки
    def deliver_materials(self, project: 'ConstructionProject') -> None:
        print(f"[Supplier] Delivering materials for project: {project.get_title()}")

    def get_role(self) -> str:
        return "Supplier"